data_structures/src/utxo_pool.rs

```diff
@@ -147,6 +150,51 @@ impl OwnUnspentOutputsPool {
     }
 }
 
+/// Struct that keep the unspent outputs pool and the own unspent outputs pool
+#[derive(Debug)]
+pub struct NodeUtxos<'a> {
+    pub all_utxos: &'a UnspentOutputsPool,
+    pub own_utxos: &'a mut OwnUnspentOutputsPool,
+}
+
+impl<'a> UtxoFantasy for NodeUtxos<'a> {
+    fn sort_iter(&self, strategy: UtxoSelectionStrategy) -> Vec<OutputPointer> {
```

Comment:
The result of this function is always used as an iterator, so we can just return an iterator here. This should allow us to remove the unneeded collect in the `Random` implementation.

```suggestion
    fn sort_iter(&self, strategy: UtxoSelectionStrategy) -> Box<dyn Iterator Item=OutputPointer> {
```

Reply:
We could open a new issue for that :P

Reply:
`UtxoFantasy` :rofl: